### PR TITLE
Fix #1480: Improve WildFly documentation and configuration

### LINKED ISSUE #1480: Improve WildFly documentation and configuration
PAS was not tested on WildFly after upgrading to Spring Boot 3. The minimal version is WildFly 27 to support EE 9.1 https://www.wildfly.org/news/2022/11/09/WildFly27-Final-Released/

Use Jave SE 17, see https://www.wildfly.org/news/2024/01/25/WildFly31-Released/

### PR DESCRIPTION
**Pull request by banterCZ**
Tested on WildFly 31.

### CODE REVIEW

File: docs/Deploying-Wildfly.md [hunk 1]
```diff
@@ -8,20 +8,11 @@ PowerAuth server contains the following configuration in `jboss-deployment-struc
 <?xml version="1.0"?>
 <jboss-deployment-structure xmlns="urn:jboss:deployment-structure:1.2">
     <deployment>
-        <exclusions>
-            <module name="org.apache.xerces" />
```
Comment (banterCZ) [inline docs/Deploying-Wildfly.md:12]: These modules are not initialized anymore.

File: docs/Deploying-Wildfly.md [hunk 2]
```diff
@@ -8,20 +8,11 @@ PowerAuth server contains the following configuration in `jboss-deployment-struc
...
-            <module name="org.apache.xalan" />
-        </exclusions>
         <exclude-subsystems>
             <!-- disable the logging subsystem because the application manages its own logging independently -->
             <subsystem name="logging" />
         </exclude-subsystems>
 
-        <resources>
-            <!-- use WAR provided Bouncy Castle -->
-            <resource-root path="WEB-INF/lib/bcprov-jdk18on-${BC_VERSION}.jar" use-physical-code-source="true"/>
```
Comment (banterCZ) [inline docs/Deploying-Wildfly.md:22]: Bouncycastle module is not initialized.

File: docs/Deploying-Wildfly.md [hunk 3]
```diff
@@ -93,12 +84,14 @@ The `application-ext.properties` file is used to override default configuration
 
 ```
 # Database Configuration - Oracle
-spring.datasource.url=jdbc:oracle:thin:@//[host]:[port]/[servicename]
```
Comment (banterCZ) [inline docs/Deploying-Wildfly.md:96]: HikariCP with `JtaTransactionManager` causes very strange behavior.

File: powerauth-admin/src/main/resources/application.properties [hunk 4]
```diff
@@ -57,4 +57,6 @@ management.tracing.sampling.probability=1.0
 #management.endpoint.prometheus.enabled=true
 #management.prometheus.metrics.export.enabled=true
 
-spring.autoconfigure.exclude=org.springframework.boot.autoconfigure.security.servlet.UserDetailsServiceAutoConfiguration
+spring.autoconfigure.exclude=\
+  org.springframework.boot.autoconfigure.security.servlet.UserDetailsServiceAutoConfiguration,\
```
Comment (banterCZ) [inline powerauth-admin/src/main/resources/application.properties:61]: It suppresses WildFly a warning about `io.micrometer.core.instrument.binder.jersey.server.MetricsApplicationEventListener`.

File: docs/Deploying-Wildfly.md [hunk 5]
```diff
@@ -8,20 +8,11 @@ PowerAuth server contains the following configuration in `jboss-deployment-struc
 <?xml version="1.0"?>
 <jboss-deployment-structure xmlns="urn:jboss:deployment-structure:1.2">
```
Comment (jnpsk) [inline docs/Deploying-Wildfly.md:9]: Maybe missing version update in doc?

```suggestion
<jboss-deployment-structure xmlns="urn:jboss:deployment-structure:1.3">
```

**Pull request merged**